plugins/device/rest.go

```diff
@@ -49,13 +54,13 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	err = txn.Ctx.Set("peerid", peerID)
 	if err != nil {
-		logger.WithError(err).Error("Failed to set data for transaction")
+		logger.WithError(err).WithField("key", peerID).Error("Failed to set key in transaction context")
```

Comment:
To understand structured logging better. Above message will be logged as 

    Failed to set key in transaction context      key=53503353-0ba8-4fd3-a8cf-aba2798ab8e5

Note: Peer ID used is just for example. 

Advantage of structured logging is that, message remains constant even if failure is for different Peer ID and all the dynamic fields are easily parsable by script. Traditionally logs are written as free hand format, but it makes very difficult to parse the logs through scripts for making sense out of log messages. Structured logging will help to address this issue.  `WithField` accepts two fields, one is key and other one is value. Key name should be self explanatory and should not consist spaces.  With this understanding, change the above log message as,

    logger.WithError(err).WithField("key", "peerid").WithField("value", peerID).Error("Failed to set key in transaction context")